src/mlpack/methods/linear_svm/linear_svm_impl.hpp

```diff
@@ -0,0 +1,165 @@
+template <typename MatType>
+LinearSVM<MatType>::LinearSVM(
+    const size_t numClasses,
+    const double lambda,
+    const double delta) :
+    numClasses(numClasses),
+    lambda(lambda),
+    delta(delta)
+{
+  // No training to do here.
+}
+
+template <typename MatType>
+template <typename OptimizerType>
+double LinearSVM<MatType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    OptimizerType optimizer)
+{
+  LinearSVMFunction<MatType> svm(data, labels,
+      numClasses, lambda, delta);
+  if (parameters.is_empty())
+    parameters = svm.InitialPoint();
+
+  // Train the model.
+  Timer::Start("linear_svm_optimization");
+  const double out = optimizer.Optimize(svm, parameters);
+  Timer::Stop("linear_svm_optimization");
+
+  Log::Info << "LinearSVM::LinearSVM(): final objective of "
+            << "trained model is " << out << "." << std::endl;
+
+  return out;
+}
+
+template <typename MatType>
+void LinearSVM<MatType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels)
+const
+{
+  arma::mat scores;
+  Classify(data, labels, scores);
+}
+
+template <typename MatType>
+void LinearSVM<MatType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels,
+    arma::mat& scores)
+const
+{
+  Classify(data, scores);
+
+  // Prepare necessary data
+  labels.zeros(data.n_cols);
+  double maxScore = 0;
+
+  // For each test input.
+  for (size_t i = 0; i < data.n_cols; ++i)
+  {
+    // For each class.
+    for (size_t j = 0; j < numClasses; ++j)
+    {
+      // If a higher class probability is encountered, change score.
+      if (scores(j, i) > maxScore)
+      {
+        maxScore = scores(j, i);
+        labels(i) = j;
+      }
+    }
+
+    // Set maximum probability to zero for next input.
+    maxScore = 0;
+  }
+}
+
+template <typename MatType>
+void LinearSVM<MatType>::Classify(
+    const MatType& data,
+    arma::mat& scores)
+const
+{
+  if (data.n_rows != FeatureSize())
+  {
+    std::ostringstream oss;
+    oss << "LinearSVM::Classify(): dataset has " << data.n_rows
+        << " dimensions, but model has " << FeatureSize() << " dimensions!";
+    throw std::invalid_argument(oss.str());
+  }
+
+  scores = parameters * data;
+}
+
+template <typename MatType>
+template <typename VecType>
+size_t LinearSVM<MatType>::Classify(const VecType& point) const
+{
+  arma::Row<size_t> label(1);
+  Classify(point, label);
+  return size_t(label(0));
+}
+
+template <typename MatType>
+double LinearSVM<MatType>::ComputeAccuracy(
+    const MatType& testData,
+    const arma::Row<size_t>& testLabels)
+const
+{
+  arma::Row<size_t> labels;
+
+  // Get predictions for the provided data.
+  Classify(testData, labels);
+
+  // Increment count for every correctly predicted label.
+  size_t count = 0;
+  for (size_t i = 0; i < labels.n_elem ; i++)
+    if (testLabels(i) == labels(i))
+      count++;
+
+  // Return the accuracy.
+  return (double)count / labels.n_elem;
```

Comment:
Are you sure this is between 0 and 100?  This looks like it is between 0 and 1.

Reply:
I updated it to `0` and `1` since the last time you pointed this out (also updated the tests accordingly).